src/main/java/executor/service/service/stepexecution/impl/StepExecutionFabricImpl.java

```diff
@@ -0,0 +1,32 @@
+package executor.service.service.stepexecution.impl;
+
+import executor.service.exceptions.StepExecutionException;
+import executor.service.model.scenario.StepAction;
+import executor.service.service.stepexecution.StepExecution;
+import executor.service.service.stepexecution.StepExecutionFabric;
+
+import java.util.List;
+
+public class StepExecutionFabricImpl implements StepExecutionFabric {
```

Comment:
Design patterns are not dogma or religion, you must understand that the common use of them one-on-one is harmful and unsustainable.
When you use pattern AbstractFactory do you know what exactly class create new object, and do you sure that factory always create new objects? I think in this way:
In our project ObjectFactory is responsible for creating new services, so I don't need to create objects, but only contain them, and return necessary on call. Or if it's clearer to your for understand StepExecutionFactory ask ObjectFactory to create StepExecutions' implemenations on bootstrap.

If you want to use pattern Strategy you must understand what is a subject of this pattern. Subject of strategy is going be step because he executes in different ways. So how do you want apply pattern Strategy to ExecutorService?

I try to set an example, in which option pattern "Strategy" must to use, when object has a general behavior, but different ways of implementaion:

One man work on delivery, he has a car, a bike and on foot, so he can do delivery in different ways.
In one day he delivery package on the car, on bike or on foot, how his boss says him to do delivery. Or when we need to print check or send it to mail, or one of messenger, we also can use Strategy. 

But our fabric injects in ExecutorService and ExecutorService choose for each step necessary implementation, but not step executes by one of the ways.

Reply:
> You complicate the readability of the code with such class names. Titles should convey the essence.

When you use getStepExecutor is it return you object?